src/utils/tokens.ts

```diff
@@ -153,6 +172,353 @@ const tokens = {
+      createMint?: boolean,
+      createMelt?: boolean,
+      nftData?: string | null,
+    } = {},
+  ): Promise<IDataTx> {
+    const isCreateNFT = !!nftData;
+    const mintOptions = {
+      createAnotherMint: createMint,
+      createMelt,
+      changeAddress,
+      isCreateNFT,
+    };
+
+    const txData = await this.prepareMintTxData(address, mintAmount, storage, mintOptions);
+    if (isCreateNFT) {
+      if (nftData === null) {
+        // This should never happen since isNFT is true only if nftData is not null
+        // But the typescript compiler doesn't seem to understand that
+        throw new Error('this should not happen');
+      }
+      // After the transaction data is completed
+      // if it's an NFT I must add the first output as the data script
+      // For NFT data the value is always 0.01 HTR (i.e. 1 integer)
+      txData.outputs.unshift({
+        type: 'data',
+        data: nftData,
+        value: 1,
+        token: HATHOR_TOKEN_CONFIG.uid,
+        authorities: 0,
+      });
+    }
+
+    // Set create token tx version value
+    txData.version = CREATE_TOKEN_TX_VERSION;
+    txData.name = name;
+    txData.symbol = symbol;
+
+    return txData;
+  },
+
+  /**
+   * Prepare delegate authority transaction data
+   *
+   * @param {string} token Token to delegate authority
+   * @param {IDataInput} authorityInput Utxo to spend
+   * @param {string} address Address to send the authority
+   * @param {IStorage} storage Storage instance of the wallet
+   * @param {boolean} [createAnother=true] If we should create another authority in the current wallet
+   * @returns {Promise<IDataTx>} Transaction data
+   */
+  async prepareDelegateAuthorityTxData(
+    token: string,
+    authorityInput: IDataInput, // Authority input
+    address: string,
+    storage: IStorage,
+    createAnother: boolean = true,
+  ): Promise<IDataTx> {
+    const outputs: IDataOutput[] = [{
+      type: getAddressType(address, storage.config.getNetwork()),
+      address,
+      token,
+      authorities: authorityInput.authorities,
+      value: authorityInput.value,
+      timelock: null,
+    }];
+
+    if (createAnother) {
+      const newAddress = await storage.getCurrentAddress();
+      outputs.push({
+        type: getAddressType(newAddress, storage.config.getNetwork()),
+        address: newAddress,
+        token,
+        authorities: authorityInput.authorities,
+        value: authorityInput.value,
+        timelock: null,
+      });
+    }
+
+    return {
+      outputs,
+      inputs: [authorityInput],
+      tokens: [token],
+    };
+  },
+
+  /**
+   * Prepare transaction data to destroy authority utxos
+   *
+   * @param authorityInputs Authority inputs to destroy
+   * @returns {IDataTx} Transaction data
+   */
+  prepareDestroyAuthorityTxData(
+    authorityInputs: IDataInput[], // Authority inputs
+  ): IDataTx {
+    return {
+      inputs: authorityInputs,
+      outputs: [],
+      tokens: [],
```

Comment:
It is weird, but this is actually great format since the `tokens` array can only be referenced by the `token_data` on the outputs we do not add unnecessary data when the custom token only exists on the inputs.